runtime/src/main/kotlin/kotlin/native/concurrent/WorkerBoundReference.kt

```diff
@@ -68,4 +82,10 @@ public class DisposableWorkerBoundReference<out T : Any>(value: T) {
      */
     val value: T
         get() = ref.value?.value ?: throw IllegalStateException("illegal attempt to dereference disposed $this")
+
+    /**
+     * The referenced value or null if [this] is disposed, or if referred object is not frozen and current worker is different from the one created [this]
+     */
+    val valueOrNull: T?
```

Comment:
@elizarov do you need a middleground `valueOrNull` that fails with exception if `this` is disposed, but returns `null` if called on the wrong worker?

Reply:
Yes. It is not that much useful without `worker` reference, though, since I'll have to store a worker reference myself anyway, but it will have me from having to use `currentWorker` and compare myself.

Reply:
> do you need a middleground valueOrNull that fails with exception if this is disposed, but returns null if called on the wrong worker?

I'm not sure that current behaviour is required. So this "middleground" should probably be the only variant.

Reply:
>  It is not that much useful without worker reference

I'll be adding `worker` getter in a later PR.

> I'm not sure that current behaviour is required. So this "middleground" should probably be the only variant.

@elizarov what do you think?

Reply:
There's also an issue with naming. It feels strange to me, if `valueOrNull` starts throwing exceptions. Maybe `valueOrNullIfSet`? @ilya-g what do you think?

Reply:
First about throwing ISE from this property:
- if using WBR after disposal is expected to be a normal case, then I'd prefer this property not to throw.
- if that's an exceptional case, then it's ok to throw an exception.
Then about naming:
- if that's an exceptional case so that the property should not be accessed in a normal code, then it's ok to leave the name `valueOrNull`
- only if that is a normal case (to access the property after disposal), and you insist on throwing exception in this case, then the property should be named differently.

Reply:
The whole `DisposableWBR` does not look very useful to me anyway. It does not need any integration with GC or anything. For convenience and easy-to-troubleshoot exceptions I'll have to write it myself anyway. I have a radical suggestion to drop `DisposableWBR` variant completely which completely avoid the needless design discussion on its behavior.

Reply:
Okay, let's remove it then.